addons/website/controllers/main.py

```diff
@@ -309,6 +311,181 @@ def get_dynamic_snippet_templates(self, filter_name=False):
+                        if field_meta.get('match') and value and term:
+                            pattern = '(%s)' % '|'.join([re.escape(search_term) for search_term in term.split(' ')])
+                            parts = re.split(pattern, value, flags=re.I)
+                            if len(parts) > 1:
+                                value = request.env['ir.ui.view'].sudo()._render_template(
+                                    "website.search_text_with_highlight",
+                                    {'parts': parts}
+                                ).decode('utf-8')
+                                field_type = 'html'
+
+                    if (field_type in ('image', 'binary')):
+                        value = escape(value)
+                    elif ('ir.qweb.field.%s' % field_type) in request.env:
+                        opt = {}
+                        if field_type == 'monetary':
+                            opt['display_currency'] = options['display_currency']
+                        elif field_type == 'html':
+                            opt['template_options'] = {}
+                        value = request.env[('ir.qweb.field.%s' % field_type)].value_to_html(value, opt)
+                    else:
+                        value = escape(value)
+                else:
+                    value = ''
+                mapped[mapped_name] = value
+            result.append(mapped)
+
+        return {
+            'results': result,
+            'results_count': results_count,
+            'parts': {key: True for mapping in mappings for key in mapping.keys()},
+        }
+
+    @http.route(['/pages', '/pages/page/<int:page>'], type='http', auth="public", website=True, sitemap=False)
+    def pages_list(self, page=1, search='', **kw):
+        Page = request.env['website.page']
+
+        domain = request.website.website_domain()
+        domain += [('website_published', '=', True)]
+        if search:
+            domain += ['|', ('name', 'ilike', search), ('url', 'ilike', search)]
+
+        pages = Page.sudo().search(domain, order='name asc, website_id desc, id')
+        pages = pages.filtered(pages._is_most_specific_page)
+        pages_count = len(pages)
+
+        step = 50
+        pager = portal_pager(
+            url="/pages",
+            url_args={'search': search},
+            total=pages_count,
+            page=page,
+            step=step
+        )
+
+        pages = pages[(page - 1) * step:page * step]
+
+        values = {
+            'pager': pager,
+            'pages': pages,
+            'search': search,
+            'search_count': pages_count,
+        }
+        return request.render("website.list_website_public_pages", values)
+
+    @http.route(['/hybrid', '/hybrid/page/<int:page>'], type='http', auth="public", website=True, sitemap=False)
+    def hybrid_list(self, page=1, search='', **kw):
+        if not search:
+            return request.render("website.list_hybrid")
```

Comment:
By clicking on "All Results" with an empty input, form submit will return empty result in "/hybrid" which is incompatible.

![img-i](https://user-images.githubusercontent.com/25032876/114898333-49d18580-9e12-11eb-99d3-925e33449bd3.png)